File: design-patterns/java/java-observer-design-pattern/Followed.java
import java.util.ArrayList;
import java.util.List;

public class Followed {
    protected List<Follower> followerList;
    protected String name;

    public Followed(String name) {
        this.name = name;
        followerList = new ArrayList<>();
    }

    public void addFollower(Follower follower) {
        followerList.add(follower);
    }

    public void removeFollower(Follower follower) {
        followerList.remove(follower);
    }

    public void notifyAll(Post post) {
        for (Follower follower : followerList) {
            follower.notify(post);
        }
    }

    public String getName() {
        return name;
    }
}

File: design-patterns/csharp/csharp-observer-design-pattern/Followed.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace csharp_observer_design_pattern
{
    public class Followed
    {
        protected List<Follower> followerList;
        protected string name;

        public Followed(string name){
            this.name = name;
            followerList = new List<Follower>();
        }

        public void AddFollower(Follower follower){
            followerList.Add(follower);
        }

        public void RemoveFollower(Follower follower){
            followerList.Remove(follower);
        }

        public void NotifyAll(Post post){
            foreach (Follower follower in followerList){
                follower.Notify(post);
            }
        }

        public string Name{
            get { return name; }
        }
        
    }
}